zorg/buildbot/reporters/utils.py

```diff
@@ -109,14 +136,156 @@ def buildAdditionalContext(self, master, ctx):
+        # Check for mode == "problem" only.
+        if results == FAILURE:
+            prev_build = build["prev_build"]
+            if (
+                prev_build and prev_build["results"] == SUCCESS
+            ):  # Note != FAILURE in base
+                return True
+        return False
+
+    @defer.inlineCallbacks
+    def generate(self, master, reporter, key, build):  # override
+        yield getDetailsForBuild(
+            master,
+            build,
+            want_properties=True,
+            want_steps=self.formatter.want_steps,
+            want_previous_build=True,
+            want_logs=self.formatter.want_logs,
+            want_logs_content=self.formatter.want_logs_content,
+        )
+        buildid = build["buildid"]
+
+        if not self.is_message_needed_by_props(build):
+            # log.msg(f"LLVMFailBuildGenerator.generate(buildid={buildid}): INFO: Not is_message_needed_by_props. Ignore.")
+            return None
+        if not self.is_message_needed_by_results(build):
+            # log.msg(f"LLVMFailBuildGenerator.generate: INFO(buildid={buildid}): INFO: Not is_message_needed_by_results. Ignore.")
+            return None
+
+        changes = yield master.data.get(("builds", buildid, "changes"))
+        log.msg(
+            f"LLVMFailBuildGenerator.generate(buildid={buildid}): INFO: changes={changes}"
+        )
+
+        len_changes = len(changes)
+        if len_changes < 1:
+            log.msg(
+                f"LLVMFailBuildGenerator.generate(buildid={buildid}): WARNING: No changes (unknown). Ignore."
+            )
+            return None
+        elif len_changes > 1:
+            log.msg(
+                f"LLVMFailBuildGenerator.generate(buildid={buildid}): WARNING: Number of changes ({len_changes}) must be 1. Ignore."
+            )
+            return None
+
+        change = changes[0]
+        # Get the first line of the commit description.
+        title = change["comments"].split("\n")[0]
+        # And store it in properties as 'title'.
+        build["properties"]["title"] = (
+            title,
+            None,
+        )
+
+        report = yield self.build_message(self.formatter, master, reporter, build)
+        # log.msg(f"LLVMFailBuildGenerator.generate(buildid={buildid}): INFO: report={report}")
+        return report
+
+
+class LLVMFailGitHubReporter(GitHubCommentPush):
+    name = "LLVMFailGitHubReporter"
+
+    def _extract_issue(self, props):  # override
+        log.msg(f"LLVMFailGitHubReporter._extract_issue: INFO: props={props}")
+        title = props.getProperty("title")
+        if title:
+            # Search for PR# in the first line of the commit description, which looks like 'Some text (#123)'.
+            m = re.search(r"^.* \(#(\d+)\)$", title)
+            if m:
+                return m.group(1)
+        return None
+
+    # Just to comment out _http.post() and add logs.
+    # TODO: Can be removed in the final version.
```

Comment:
It wold be nice to have rationale in this comment, for clarity of when it could be removed and what are the dependencies.